react/src/views/CreateBox/CreateBox.test.js

```diff
@@ -0,0 +1,270 @@
+  let component = null;
+  beforeEach(() => {
+    const history = createMemoryHistory();
+    const state = { qr: "barcode=387b0f0f5e62cebcafd48383035a92a" };
+    history.push("/create-box", state);
+
+    component = render(
+      <MockedProvider mocks={mocks} addTypename={false}>
+        <Router history={history}>
+          <CreateBox />
+        </Router>
+      </MockedProvider>,
+    );
+  });
+
+  afterEach(cleanup);
+
+  it("After a successful submission, the form disappears and a screen with `You created a new box` appears with a box-id", async () => {
+    const submitBtn = component.getByRole("button", { name: /do the mutation/i });
+
+    fireEvent.click(submitBtn);
+
+    await waitFor(() => {
+      expect(component.queryByTestId("createBoxForm")).toBeNull();
+      expect(component.getByTestId("createdBox")).toBeTruthy();
+      expect(component.getByRole("heading", { name: /the box id is: 456/i })).toHaveTextContent(
+        "The Box ID is: 456",
+      );
+    });
+  });
+});
+
+// Need to create an error message for blank required form fields
+describe("Required form fields prohibit submission when blank", () => {
+  let component = null;
+  beforeEach(() => {
+    const history = createMemoryHistory();
+    const state = { qr: "barcode=387b0f0f5e62cebcafd48383035a92a" };
+    history.push("/create-box", state);
+
+    component = render(
+      <MockedProvider mocks={mocks} addTypename={false}>
+        <Router history={history}>
+          <CreateBox />
+        </Router>
+      </MockedProvider>,
+    );
+  });
+
+  afterEach(cleanup);
+
+  it("does nothing when locationId, productId, items, and sizeId are blank", async () => {
+    const inputFields = component.getAllByRole("spinbutton");
+    const submitBtn = component.getByRole("button", { name: /do the mutation/i });
+
+    for (let i = 0; i < inputFields.length; i++) {
+      fireEvent.change(inputFields[i], { target: { value: "" } });
+    }
+
+    await waitFor(() => {
+      for (let i = 0; i < inputFields.length; i++) {
+        expect(inputFields[i].value).toBe("");
+      }
+    });
+
+    fireEvent.click(submitBtn);
+
+    await waitFor(() => {
+      // setTimeout compensates for lack of an error message
+      setTimeout(() => {
+        // expect an error message to appear (e.g. "Please fill out required form fields")
+        expect(component.queryByTestId("createBoxForm")).toBeTruthy();
+        expect(component.queryByTestId("createdBox")).toBeNull();
+        expect(component.queryByTestId("loadingState")).toBeNull();
+        expect(component.queryByTestId("errorState")).toBeNull();
+      }, 5000);
+    });
+  });
+});
+
+describe("Network error after submission", () => {
+  let component = null;
+  beforeEach(() => {
+    const history = createMemoryHistory();
+    const state = { qr: "barcode=387b0f0f5e62cebcafd48383035a92a" };
+    history.push("/create-box", state);
+
+    component = render(
+      <MockedProvider
+        mocks={[
```

Comment:
this should be defined in the beginning of the doc and passed into a utility function